cmd/disk-cache-config.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// cacheConfig represents cache configuration
+type cacheConfig struct {
+	sync.RWMutex
+	Drives  []string `json:"drives"`
+	Expiry  int      `json:"expiry"`
+	Exclude []string `json:"exclude"`
+}
+
+// CacheConfig represents cache config settings
+type CacheConfig struct {
+	Drives  []string
+	Expiry  int
+	Exclude []string
+}
+
+// GetDrives gets the list of cache disk paths
+func (c *cacheConfig) GetDrives() []string {
+	c.RLock()
+	defer c.RUnlock()
+	if len(c.Drives) > 0 {
+		globalIsDiskCacheEnabled = true
+	}
+	c2 := make([]string, len(c.Drives))
+	for _, d := range c.Drives {
+		c2 = append(c2, d)
+	}
+	return c2
+}
+
+// GetExpiry gets the expiry time for stale entries in the cache
+func (c *cacheConfig) GetExpiry() int {
+	c.RLock()
+	defer c.RUnlock()
+	return c.Expiry
+}
+
+// GetExcludes gets the list of wildcards to exclude from caching.
+func (c *cacheConfig) GetExcludes() []string {
+	c.RLock()
+	defer c.RUnlock()
+	c2 := make([]string, len(c.Exclude))
+	for _, e := range c.Exclude {
+		c2 = append(c2, e)
+	}
+	return c2
+}
+
+// GetConfig() returns config settings
+func (c *cacheConfig) GetConfig() (config CacheConfig) {
+	c.RLock()
+	defer c.RUnlock()
+
+	d := c.Drives
+	if len(c.Drives) == 0 {
+		d = []string{}
+	}
+	e := c.Exclude
+	if len(c.Exclude) == 0 {
+		e = []string{}
+	}
+
+	config = CacheConfig{Expiry: c.Expiry,
+		Drives:  d,
+		Exclude: e,
+	}
+	return
+}
+
```

Comment:
Looks like the methods on cacheConfig are not used. So you can remove them. You can remove cacheConfig too I think?